tests/RobotFramework/tests/cumulocity/registration/device_deregistration.robot

```diff
@@ -0,0 +1,103 @@
+*** Settings ***
+Resource    ../../../resources/common.resource
+Library    Cumulocity
+Library    ThinEdgeIO
+
+Test Tags    theme:c8y    theme:registration
+Suite Setup    Custom Setup
+Test Setup    Test Setup
+Test Teardown    Get Logs    ${DEVICE_SN}
+
+*** Test Cases ***
+
+Main device registration
+    ${mo}=    Device Should Exist              ${DEVICE_SN}
+    ${mo}=    Cumulocity.Device Should Have Fragment Values    name\=${DEVICE_SN}
+    Should Be Equal    ${mo["owner"]}    device_${DEVICE_SN}
+    Should Be Equal    ${mo["name"]}    ${DEVICE_SN}
+
+
+Child device registration
+    Execute Command    tedge mqtt pub --retain 'te/device/${CHILD_SN}//' '{"@type":"child-device","@id":"${CHILD_SN}"}'
+
+    # Check registration
+    ${child_mo}=    Device Should Exist        ${CHILD_SN}
+    ${child_mo}=    Cumulocity.Device Should Have Fragment Values    name\=${CHILD_SN}
+    Should Be Equal    ${child_mo["owner"]}    device_${DEVICE_SN}    # The parent is the owner of the child
+    Should Be Equal    ${child_mo["name"]}     ${CHILD_SN}
+
+    # Check child device relationship
+    Cumulocity.Set Device    ${DEVICE_SN}
+    Cumulocity.Should Be A Child Device Of Device    ${CHILD_SN}
+
+    #Deregister Child device
+    Run Keyword And Ignore Error    Execute Command    mosquitto_sub --remove-retained -W 3 -t 'te/device/${CHILD_SN}//'
+    
+
+Register child device with defaults via MQTT
+    Execute Command    tedge mqtt pub --retain 'te/device/${CHILD_SN}//' '{"@type":"child-device"}'
+    Check Child Device    parent_sn=${DEVICE_SN}    child_sn=${CHILD_XID}    child_name=${CHILD_XID}    child_type=thin-edge.io-child
+
+    #Deregister Child device
+    Run Keyword And Ignore Error    Execute Command    mosquitto_sub --remove-retained -W 3 -t 'te/device/${CHILD_SN}//'
+
+Register service on a child device via MQTT
+    Execute Command    tedge mqtt pub --retain 'te/device/${CHILD_SN}//' '{"@type":"child-device","name":"${CHILD_SN}","type":"linux-device-Aböut"}'
+    Execute Command    tedge mqtt pub --retain 'te/device/${CHILD_SN}/service/custom-app' '{"@type":"service","@parent":"device/${CHILD_SN}//","name":"custom-app","type":"custom-type"}'
+
+    # Check child registration
+    Check Child Device    parent_sn=${DEVICE_SN}    child_sn=${CHILD_XID}    child_name=${CHILD_SN}    child_type=linux-device-Aböut
+
+    # Check service registration
+    Check Service    child_sn=${CHILD_XID}    service_sn=${CHILD_XID}:service:custom-app    service_name=custom-app    service_type=custom-type    service_status=up
+
+    #Deregister service on a child device
+    Run Keyword And Ignore Error    Execute Command    mosquitto_sub --remove-retained -W 3 -t 'te/device/${CHILD_SN}/service/custom-app/#'
+
+
```

Comment:
Deregistration of services can also be tested the same way as described above for devices, by deleting the service and recreating it, which should succeed.